### PR TITLE
Remove trailing DTO from object name in validation messages

### PR DESCRIPTION
**Pull request by yelhouti**
Fix https://github.com/jhipster/generator-jhipster/issues/10049

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [X] Tests are added where necessary
-   [X] Documentation is added/updated where necessary
-   [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/main/java/package/web/rest/errors/ExceptionTranslator.java.ejs [hunk 1]
```diff
@@ -137,7 +137,7 @@ _%>
     public <%- returnType %> handle<%_ if (!reactive) { _%>MethodArgumentNotValid(MethodArgumentNotValidException<% } else { %>BindingResult(WebExchangeBindException<% } %> ex, @Nonnull <%= requestClass %> request) {
         BindingResult result = ex.getBindingResult();
         List<FieldErrorVM> fieldErrors = result.getFieldErrors().stream()
-            .map(f -> new FieldErrorVM(f.getObjectName(), f.getField(), f.getCode()))
+            .map(f -> new FieldErrorVM(f.getObjectName().replaceFirst("DTO$", ""), f.getField(), f.getCode()))
```
Comment (antarus) [inline generators/server/templates/src/main/java/package/web/rest/errors/ExceptionTranslator.java.ejs]: Use dtoSuffix variable instead of DTO ?

Comment (yelhouti) [reply generators/server/templates/src/main/java/package/web/rest/errors/ExceptionTranslator.java.ejs]: Indeed, thanks @antarus !

**Comment by DanielFran** [comment]
Closed/Reopen to launch testa again

**Comment by pascalgrimaud** [comment]
The builds failed with this stacktrace:

```
[ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.393 s <<< FAILURE! - in io.github.jhipster.sample.web.rest.errors.ExceptionTranslatorIT
[ERROR] testMethodArgumentNotValid  Time elapsed: 0.167 s  <<< FAILURE!
java.lang.AssertionError: JSON path "$.fieldErrors.[0].objectName" expected:<test> but was:<testDTO>
	at io.github.jhipster.sample.web.rest.errors.ExceptionTranslatorIT.testMethodArgumentNotValid(ExceptionTranslatorIT.java:58)
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ExceptionTranslatorIT.testMethodArgumentNotValid:58 JSON path "$.fieldErrors.[0].objectName" expected:<test> but was:<testDTO>
[INFO] 
[ERROR] Tests run: 686, Failures: 1, Errors: 0, Skipped: 0
```

Can you have a look plz ?

**Comment by yelhouti** [comment]
I have been struggling for two days to understand the problem, since after running the same tests on my machine it works. I tried everything, can anyone help with this?

**Comment by pascalgrimaud** [comment]
@yelhouti : I'll have a look on this today

**Comment by yelhouti** [comment]
@pascalgrimaud thanks a lot :)

**Comment by ruddell** [comment]
The failing test has a DTO suffix of `""` ([`.yo-rc.json`](https://github.com/jhipster/generator-jhipster/blob/7179b7a657df8ff0ebc2f0797ba943b6d241ebe7/test-integration/samples/ngx-default/.yo-rc.json#L73)), so `"testDTO".replace("", "")` results in `testDTO` (not the expected `test`).  

Maybe use the DTO suffix in the test instead of hardcoding testDTO?

**Comment by pascalgrimaud** [comment]
@yelhouti : just push directly to your branch. hope it should be ok now

**Comment by yelhouti** [comment]
@pascalgrimaud thanks a lot :), nice spot @ruddell

**Comment by yelhouti** [comment]
Thank you team for the help :) . Here is my first bounty claim! https://opencollective.com/generator-jhipster/expenses/9611

**Pull request merged**